File: atcoder/5209.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;
const ll mod = 1e9 + 7;
ll pre, suf;
int main()
{
    int n, k;
    scanf("%d%d", &n, &k);
    ll ans = 0;
    for (int i = k; i <= n + 1; ++i)
    {
        pre = i * (i - 1ll) / 2;
        suf = (n * 2 + 1ll - i) * i / 2;
        ans = (ans + suf - pre + 1) % mod;
    }
    printf("%lld\n", ans);
    return 0;
}

File: atcoder/5209.lua
local n, k = io.read('*n', '*n')
local mfl =  math.floor
local mod = 1000000007
local ans = 0
for i = k, n + 1 do
    local pre = mfl(i * (i - 1) / 2)
    local suf = mfl((n + n - i + 1) * i / 2)
    ans = (ans + suf - pre + 1) % mod
end
io.write(ans)